### DISCUSSION TITLE
How to change from Tasmota to this firmware

### DISCUSSION THREAD

**Discussion by g0r3** [mongoose-os-apps/shelly-homekit#1441] [Q&A] [answered]
Hi,

since quite a while I am researching on how to revert from Tasmota on a Shelly Plus Plug S to this firmware. Is there a way that I can flash this firmware over the air and doesn't involve opening up the device to get access to the serial interface?

Thanks!

**Comment by markirb** [comment, marked as answer]
Unfortunately the only answer is a serial interface. 

Though it is possible you would need to write a firmware that repartitions the device and afterwards flashes it. Tasmota does the repartitioning but only in the other direction.

**Reply by timoschilling** [reply]
I think that's not true, there is a different way as well. I have never tested it and I we can't give support for it but you can take a look at https://github.com/yaourdt/tasmota-to-mgos

**Reply by markirb** [reply]
Ah did not know that. This could work yes. However does not seem to be maintained and does not support esp32

**Reply by timoschilling** [reply]
FTR: no esp32 means no Shelly Plus, Pro, Mini
